src/status.rs

```diff
@@ -101,103 +101,124 @@ pub async fn serve(
             continue;
         }
 
-        // Hijack server status packet
-        if client_state == ClientState::Status && packet.id == packets::status::SERVER_STATUS {
-            let server_status = server_status(&client_info, &config, &server).await;
-            let packet = StatusResponse { server_status };
-
-            let mut data = Vec::new();
-            packet.encode(&mut data).map_err(|_| ())?;
-
-            let response = RawPacket::new(0, data).encode_with_len(&client)?;
-            writer.write_all(&response).await.map_err(|_| ())?;
-
-            continue;
-        }
-
         // Hijack ping packet
-        if client_state == ClientState::Status && packet.id == packets::status::SERVER_PING {
+        if client_state == ClientState::Status && packet.id == packets::status::SERVER_PING
+        {
             writer.write_all(&raw).await.map_err(|_| ())?;
             continue;
         }
 
-        // Hijack login start
-        if client_state == ClientState::Login && packet.id == packets::login::SERVER_LOGIN_START {
-            // Try to get login username, update client info
-            // TODO: we should always parse this packet successfully
-            let username = LoginStart::decode(&mut packet.data.as_slice())
-                .ok()
-                .map(|p| p.name);
-            client_info.username = username.clone();
-
-            // Kick if lockout is enabled
-            if config.lockout.enabled {
-                match username {
-                    Some(username) => {
-                        info!(target: "lazymc", "Kicked '{}' because lockout is enabled", username)
-                    }
-                    None => info!(target: "lazymc", "Kicked player because lockout is enabled"),
-                }
-                action::kick(&client, &config.lockout.message, &mut writer).await?;
-                break;
-            }
+        // Grab the server and config
+        if let Some(handshake) = client_info.handshake.as_ref() {
+            if let Some((config, server)) = servers.get(&handshake.server_addr) {
```

Comment:
These two lines look like code smell, is there a more elegant way to do this?

Reply:
You could probably do this:

```diff
-        if let Some(handshake) = client_info.handshake.as_ref() {
-            if let Some((config, server)) = servers.get(&handshake.server_addr) {
+        if let Some((config, server)) = client_info
+            .handshake
+            .as_ref()
+            .and_then(|handshake| servers.get(&handshake.server_addr))
+        {
```

and then remove the extra `}`.

Reply:
Thank you, implemented